akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -86,6 +86,33 @@ private[io] final class AsyncDnsResolver(
     result
   }
 
+  private def resolveWithSearch(name: String, requestType: RequestType, resolver: ActorRef): Future[DnsProtocol.Resolved] = {
+    if (settings.SearchDomains.nonEmpty) {
+      val nameWithSearch = settings.SearchDomains.map(sd ⇒ name + "." + sd)
```

Comment:
what is one of these doesn't have enough dots?

Reply:
Originally I was just using a filter, but after rereading the documentation, it makes it clear that this only applies to the name being searched, see [here](http://man7.org/linux/man-pages/man5/resolv.conf.5.html):

> Sets a threshold for the number of dots which must
> appear in a name given to res_query(3) (see
> resolver(3)) before an initial absolute query will be
> made.  The default for n is 1, meaning that if there
> are any dots in a name, the name will be tried first as
> an absolute name before any search list elements are
> appended to it.  The value for this option is silently
> capped to 15.

The "initial" I think is the key there, ndots only affects the initial query, not subsequent queries based on search domains. Furthermore, the logic here is the same as the logic used by netty.

Now, I've just done a test on my linux box, and it's hard to work out exactly what it's doing, but for example if I set:

```
search com
options ndots:2
```

Then `host github` gives me `github.com`, so `ndots:2` isn't preventing `github.com` to be searched. But, if I search `host github.com`, then I get `github.com.com`, (which redirects to `com.com` and is a site setup to capture accidental double coms in domain names and try and monetise that traffic). Interestingly, if I search `host akka.io`, I get `akka.io`. So presumably, `akka.io.com` was searched (it doesn't resolve), so it then tried `akka.io`. So it appears that the implementation in Linux is actually not quite as the documentation states, rather, ndots controls whether the hostname passed in is searched first or last. We could use that behaviour too, and I think that would have the effect of fixing (or at least, ensuring that the old behaviour is still implemented) the `localhost` issue.

Reply:
I've modified the code now to put the host at the end of the search list if it fails the ndots test, and I also added a test to ensure resolving `localhost` works, which passes because the nameserver we use in the integration test resolves `localhost`. See https://github.com/akka/akka/compare/a02a6375d638eee1864cdf0a0c080d57c16f261f..e75212eed78b48a82457d66252c6861c32ad577d for changes.

Reply:
:+1: